internal/services/managementgroup/management_group_data_source.go

```diff
@@ -41,6 +41,11 @@ func dataSourceManagementGroup() *pluginsdk.Resource {
 				ExactlyOneOf: []string{"name", "display_name"},
 			},
 
+			"management_group_id": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
```

Comment:
(as elsewhere) could we call this `tenant_scoped_id`? I'm omitting `management_group` in the field name here since it's in the resource type